lm-composed/etc/populate-lm-api.py

```diff
@@ -0,0 +1,78 @@
+#!/app/lm-agent/.venv/bin/python
+import os
+import httpx
+
+
+LM_AGENT_OIDC_DOMAIN = os.getenv("LM_AGENT_OIDC_DOMAIN")
+LM_AGENT_OIDC_CLIENT_ID = os.getenv("LM_AGENT_OIDC_CLIENT_ID")
+LM_AGENT_OIDC_CLIENT_SECRET = os.getenv("LM_AGENT_OIDC_CLIENT_SECRET")
+LM_AGENT_BACKEND_BASE_URL = os.getenv("LM_AGENT_BACKEND_BASE_URL")
+
+
+def get_access_token():
+    response = httpx.post(
+        f"http://{LM_AGENT_OIDC_DOMAIN}/protocol/openid-connect/token",
+        data={
+            "client_id": LM_AGENT_OIDC_CLIENT_ID,
+            "client_secret": LM_AGENT_OIDC_CLIENT_SECRET,
+            "grant_type": "client_credentials"
+        }
+    )
+    return response.json()["access_token"]
+
+
+def clean_table(token, table_name):
+    rows = httpx.get(
+        f"{LM_AGENT_BACKEND_BASE_URL}/lm/{table_name}",
+        headers={"Authorization": f"Bearer {token}"}
+    ).json()
+
+    for row in rows:
+        _id = row["id"]
+        
+        httpx.delete(
+            f"{LM_AGENT_BACKEND_BASE_URL}/lm/{table_name}/{_id}",
+            headers={"Authorization": f"Bearer {token}"}
+        )
```

Comment:
I think you should probably check the return code of these requests and throw an exception or exit with an error code if something goes wrong. Otherwise, it could be tricky to troubleshoot what's going wrong.

Reply:
Added error handling to this script, this way it will log and exit if the status code is not successful.